### PR TITLE
Add Configuration to PatternLayout

### PR DESCRIPTION
**Pull request by pgomulka**
in 2.17.2 (patch release) log4j has made a refactoring that requires a Configuration to be manually
passed into the created PatternLayout
If the Configuration is not passed, the System Variable lookup will not work
This results in cluster.name field not being populated in logs if ESv7 layout was used on ESv8 cluster (cloud)

This commit creates a PatternLayout with a DefaultConfiguration ([the same was used previous to the refactoring](https://github.com/apache/logging-log4j2/commit/9ffa643d69cdca4764bc2c4c0fe529379d17db2b#diff-0a2384aacbc3b28013b2f813d55a6504ae18d858d141a76b333606b2cdea98aaL766))
Also adds back tests that verifies ESv8 cluster with ESv7 logging config. We still have that situation in the cloud so it is better to have tests around this.

relates https://github.com/apache/logging-log4j2/issues/1551

### CODE REVIEW

**Comment by pgomulka** [comment]
we should apply the fix in 7.17 if we ever upgrade log4j to latest version there

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Comment by pgomulka** [comment]
note to reviewers: this is basically a 3line change https://github.com/elastic/elasticsearch/pull/97679/files#diff-535fddf38184fad984ddc5fce61e21774c9e12a74d69022c9627e878c190d84aL166

the rest of the code changes bring back tests that we used to have in 8.0ish - testing the 8.x cluster with 7.x config

File: qa/logging-config/src/javaRestTest/resources/plugin-security.policy [hunk 1]
```diff
@@ -0,0 +1,5 @@
+grant {
```
Comment (rjernst) [inline qa/logging-config/src/javaRestTest/resources/plugin-security.policy:1]: Why is this needed when the test has security manager disabled?

File: qa/logging-config/custom-log4j2.properties [hunk 2]
```diff
@@ -0,0 +1,187 @@
+#
```
Comment (rjernst) [inline qa/logging-config/custom-log4j2.properties:1]: Can this properties file be named to make clear it is for v7?

File: qa/logging-config/build.gradle [hunk 3]
```diff
@@ -5,8 +5,34 @@
...
+testClusters.matching { it.name == "javaRestTest" }.configureEach {
+  /**
+   * Provide a custom log4j configuration where layout is an old style pattern and confirm that Elasticsearch
+   * can successfully startup.
+   */
+  extraConfigFile 'log4j2.properties', file('custom-log4j2.properties')
+}
+
+tasks.named("javaRestTest").configure {
+  nonInputProperties.systemProperty 'tests.logfile',
```
Comment (rjernst) [inline qa/logging-config/build.gradle:30]: I think this might make sense to convert the test to the new java rest test framework. Then you won't need these system properties (you can inspect the logs from the test), you don't need to disable security manager (already done for you), and configuration of the cluster sits alongside the test itself. Just a thought.

**Review by rjernst — APPROVED** [review]
Looks good, just a couple suggestions.

File: qa/logging-config/build.gradle (hunk 3 above)
Comment (pgomulka) [reply qa/logging-config/build.gradle:30]: it was tempting to do this, but at the same time I just wanted to 'revert' the commit that removed those tests. I will create an issue to follow up on this

File: qa/logging-config/src/javaRestTest/resources/plugin-security.policy (hunk 1 above)
Comment (pgomulka) [reply qa/logging-config/src/javaRestTest/resources/plugin-security.policy:1]: this is for the IntegTests.
the ESJsonLayoutTests which has securiyt manager disabled is a unit test

File: qa/logging-config/custom-log4j2.properties (hunk 2 above)
Comment (pgomulka) [reply qa/logging-config/custom-log4j2.properties:1]: will do
